diff --git third_party/llvm/workspace.bzl third_party/llvm/workspace.bzl
index 038e0ee5fe5..4693f5cfadc 100644
--- third_party/llvm/workspace.bzl
+++ third_party/llvm/workspace.bzl
@@ -5,15 +5,15 @@ load("//third_party:repo.bzl", "tf_http_archive")
 def repo(name):
     """Imports LLVM."""
     LLVM_COMMIT = "d8415b02a519f222ecf71b069c96cc85ac635de3"
-    LLVM_SHA256 = "05fffac826b16218bbc0f882204734e68566fd0b61a1196f12c16a58b0b8af58"
+    LLVM_SHA256 = "31a8507ca4b812724adba8274747c8f69b29f78d9b0949a7ecc54ec74d31c1d4"
 
     tf_http_archive(
         name = name,
         sha256 = LLVM_SHA256,
-        strip_prefix = "llvm-project-{commit}".format(commit = LLVM_COMMIT),
+        strip_prefix = "llvm-llvm-project-{commit_partial}".format(commit_partial = LLVM_COMMIT[:7]),
         urls = [
             "https://storage.googleapis.com/mirror.tensorflow.org/github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
-            "https://github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
+            "https://api.github.com/repos/llvm/llvm-project/tarball/{commit}".format(commit = LLVM_COMMIT),
         ],
         build_file = "//third_party/llvm:llvm.BUILD",
         patch_file = [
