diff --git tensorflow_federated/BUILD tensorflow_federated/BUILD
index ff6bc8e22..bab38b2be 100644
--- tensorflow_federated/BUILD
+++ tensorflow_federated/BUILD
@@ -27,6 +27,24 @@ py_library(
         "//tensorflow_federated/python/program",
         "//tensorflow_federated/python/simulation",
         "@federated_language//federated_language",
+        "@pypi_absl_py//:pkg",
+        "@pypi_attrs//:pkg",
+        "@pypi_cachetools//:pkg",
+        "@pypi_dm_tree//:pkg",
+        "@pypi_dp_accounting//:pkg",
+        "@pypi_google_vizier//:pkg",
+        "@pypi_googleapis_common_protos//:pkg",
+        "@pypi_grpcio//:pkg",
+        "@pypi_jax//:pkg",
+        "@pypi_jaxlib//:pkg",
+        "@pypi_numpy//:pkg",
+        "@pypi_portpicker//:pkg",
+        "@pypi_scipy//:pkg",
+        "@pypi_tensorflow_model_optimization//:pkg",
+        "@pypi_tensorflow_privacy//:pkg",
+        "@pypi_tensorflow//:pkg",
+        "@pypi_tqdm//:pkg",
+        "@pypi_typing_extensions//:pkg",
     ],
 )
 
