diff --git third_party/llvm/workspace.bzl third_party/llvm/workspace.bzl
index 038e0ee5fe5..4693f5cfadc 100644
--- third_party/llvm/workspace.bzl
+++ third_party/llvm/workspace.bzl
@@ -5,15 +5,15 @@ load("//third_party:repo.bzl", "tf_http_archive")
 def repo(name):
     """Imports LLVM."""
     LLVM_COMMIT = "dc275fd03254d67d29cc70a5a0569acf24d2280d"
-    LLVM_SHA256 = "3e91127af59a6b07fea7901c80a7b8b9234eced42b0f14abbad5f9f7674dba69"
+    LLVM_SHA256 = "2f59ffcfaf44e5ced2ed57ad777704f4cfa43afbd4f6fa62f195465cc91f5964"
 
     tf_http_archive(
         name = name,
         sha256 = LLVM_SHA256,
-        strip_prefix = "llvm-project-{commit}".format(commit = LLVM_COMMIT),
+        strip_prefix = "llvm-llvm-project-{commit_partial}".format(commit_partial = LLVM_COMMIT[:7]),
         urls = [
             "https://storage.googleapis.com/mirror.tensorflow.org/github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
-            "https://github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
+            "https://api.github.com/repos/llvm/llvm-project/tarball/{commit}".format(commit = LLVM_COMMIT),
         ],
         build_file = "//third_party/llvm:llvm.BUILD",
         patch_file = [
