--- tensorflow/tsl/platform/test.h
+++ tensorflow/tsl/platform/test.patched.h
@@ -40,7 +40,6 @@
 // better error messages, more maintainable tests and more test coverage.
 #if !defined(PLATFORM_GOOGLE) && !defined(PLATFORM_GOOGLE_ANDROID) && \
     !defined(PLATFORM_CHROMIUMOS)
-#include <gmock/gmock-generated-matchers.h>  // IWYU pragma: export
 #include <gmock/gmock-matchers.h>            // IWYU pragma: export
 #include <gmock/gmock-more-matchers.h>       // IWYU pragma: export
 #endif
