--- tensorflow/python/debug/lib/BUILD
+++ tensorflow/python/debug/lib/BUILD
@@ -175,6 +175,7 @@
     deps = [
         ":profiling",
         "//third_party/py/numpy",
+        "@com_google_absl_py//absl:command_name",
     ],
 )

--- tensorflow/python/debug/lib/source_utils.py
+++ tensorflow/python/debug/lib/source_utils.py
@@ -19,7 +19,7 @@
 import re
 import zipfile

-import absl
+import absl.command_name
 import numpy as np

 from tensorflow.python.debug.lib import profiling
@@ -29,7 +29,7 @@
     os.path.dirname(os.path.dirname(os.path.dirname(
         os.path.normpath(os.path.abspath(__file__))))))

-_ABSL_BASEDIR = os.path.dirname(absl.__file__)
+_ABSL_BASEDIR = os.path.dirname(absl.command_name.__file__)


 UNCOMPILED_SOURCE_SUFFIXES = (".py")

