diff --git third_party/llvm/workspace.bzl third_party/llvm/workspace.bzl
index 038e0ee5fe5..4693f5cfadc 100644
--- third_party/llvm/workspace.bzl
+++ third_party/llvm/workspace.bzl
@@ -5,15 +5,15 @@ load("//third_party:repo.bzl", "tf_http_archive")
 def repo(name):
     """Imports LLVM."""
     LLVM_COMMIT = "10939d1d580b9d3c9c2f3539c6bdb39f408179c0"
-    LLVM_SHA256 = "4adce5ef34c2062be0d7c5eb2a11606fa70690342e7e93327457ee2b6ad7ac72"
+    LLVM_SHA256 = "8f5201fc907d4faeb6f05eaa61a24504d8685eb92e5fec143bf981783711363f"
 
     tf_http_archive(
         name = name,
         sha256 = LLVM_SHA256,
-        strip_prefix = "llvm-project-{commit}".format(commit = LLVM_COMMIT),
+        strip_prefix = "llvm-llvm-project-{commit_partial}".format(commit_partial = LLVM_COMMIT[:7]),
         urls = [
             "https://storage.googleapis.com/mirror.tensorflow.org/github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
-            "https://github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
+            "https://api.github.com/repos/llvm/llvm-project/tarball/{commit}".format(commit = LLVM_COMMIT),
         ],
         build_file = "//third_party/llvm:llvm.BUILD",
         patch_file = [
